### DISCUSSION TITLE
How to access electron methods in Quasar V2?

### DISCUSSION THREAD

**Discussion by wsw70** [quasarframework/quasar#9745] [CLI - Electron mode] [answered]
I am trying to build my first electron app with Quasar **V2** (and first electron app full stop).

How can I access electron methods such as [`BrowserWindow()`](https://www.electronjs.org/docs/latest/api/browser-window/#browserwindow-setfullscreen-flag)? 

In the V1 docs, [I saw](https://quasar.dev/quasar-cli/developing-electron-apps/node-integration) that there was a way to call them via (for example) `this.$q.electron.remote.BrowserWindow.getFocusedWindow().minimize()`. 

How to make such a call with V2?

EDIT: I managed to go fullscreen by editing `electron-main.js` but this sis just a workaround as I would like to change the viewport programmatically

```
mainWindow = new BrowserWindow({
    width: 1000,
    height: 600,
    fullscreen: true,
(...)
```

**Comment by JokerMartini** [comment]
I'm in the same boat, i don't know how to access the 'remote' module of node.js that was much easier in Quasar 1.0. I'm not sure how they do it in v2 yet.

**Reply by wsw70** [reply]
I did not find any solutions so far, I will post to other communities such as reddit or SO, maybe more luck there

**Reply by metalsadman** [reply]
have you check this section? https://quasar.dev/quasar-cli/developing-electron-apps/electron-preload-script#introduction

**Reply by rstoenescu** [reply]
Saying "it was much easier in Quasar v1.0" is not the right way to look at this. You should exclusively look at the Electron version being used and at Electron's team decisions regarding its security (because it was strengthened but this results in a bit harder way to use it).

Regardless, look for `webPreferences` in the main thread (this has to do with Electron itself). There are ways in which you can make it work like previous Electron versions (webPreferences > [contextIsolation](https://www.electronjs.org/docs/tutorial/context-isolation)) but at the expense of your app's security (which we don't recommend you to trade in). Or directly use the same old electron version with Quasar v2 also :)
**Best is to adapt to the newer Electron standards sooner than later**. This means using the preload script.

**Reply by JokerMartini** [reply]
I did not mean in any way to be rude. K think the new way of using it is absolutely the way to go in order to make safer apps.  I'm glad you are making it safer. I would suggest maybe including some more info or examples in the docs showing how to use eletrcon packages in this new framework. 
Great work as always! I'm a huge supporter of quasar.

**Reply by froschi123** [reply]
very nice discussion, but there is an answer outstanding. How do we access the getFocusedWindow function in 2.x? I used the example in the quasar documentsation under CLI and Electron, but it is not working, i get the error 

Uncaught Error: Could not call remote method 'getFocusedWindow'. Check that the method signature is correct. Underlying error: Cannot read properties of null (reading 'getHiddenValue')

How to solve this?

**Comment by rstoenescu** [comment, marked as answer]
You should exclusively look at the Electron version being used and at Electron's team decisions regarding its security (because it was strengthened but this results in a bit harder way to use it).

Regardless, look for `webPreferences` in the main thread (this has to do with Electron itself). There are ways in which you can make it work like previous Electron versions (webPreferences > [contextIsolation](https://www.electronjs.org/docs/tutorial/context-isolation)) but at the expense of your app's security (which we don't recommend you to trade in). Or directly use the same old electron version with Quasar v2 also :)

**Best is to adapt to the newer Electron standards sooner than later**. This means using [the preload script](https://quasar.dev/quasar-cli/developing-electron-apps/electron-preload-script#introduction).
